app/src/androidTest/java/com/sample/browserstack/samplecalculator/ScreenshotUtils.java

```diff
@@ -0,0 +1,85 @@
+package com.sample.browserstack.samplecalculator;
+
+import android.graphics.Bitmap;
+import android.os.Build;
+import android.os.Environment;
+import android.util.Log;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.runner.screenshot.BasicScreenCaptureProcessor;
+import androidx.test.runner.screenshot.ScreenCapture;
+import androidx.test.runner.screenshot.Screenshot;
+
+import java.io.File;
+import java.io.IOException;
+
+public class ScreenshotUtils {
+
+    private final String methodName;
+    private final String className;
+    private final CustomScreencaptureProcessor customProcesssor;
+
+    ScreenshotUtils() {
+        StackTraceElement testClass = findTestClassTraceElement(Thread.currentThread().getStackTrace());
+        className = testClass.getClassName().replaceAll("[^A-Za-z0-9._-]", "_");
+        methodName = testClass.getMethodName();
+        customProcesssor = new CustomScreencaptureProcessor();
+    }
+
+    public void captureScreenshot(String screenshotName) {
+        ScreenCapture capture = Screenshot.capture();
+        capture.setFormat(Bitmap.CompressFormat.PNG);
+        capture.setName(screenshotName);
+
+        try {
+            customProcesssor.process(capture);
+        } catch (IOException e) {
```

Comment:
this should `error` the test

Reply:
Yes, added the `throw new Exception` logic in the catch block which will ensure the same.